src/main/java/com/taakestan/video/VideoTransformer.java

```diff
@@ -0,0 +1,95 @@
+package com.taakestan.video;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.OS;
+import org.apache.commons.exec.PumpStreamHandler;
+
+/**
+ * @author Omid Pourhadi : omidpourhadi [AT] gmail [DOT] com
+ *
+ */
+public class VideoTransformer
+{
+
+    private static final Logger log = Logger.getLogger(VideoTransformer.class.getName());
+
+    
+    /**
+     * @param fielPath : full file path in filesystem
+     * @param outputDirectory : 
+     * @return : null or outputdirectory + fileName_web.mp4 file
+     */
+    public static File transform(String fielPath, String outputDirectory)
+    {
+        if (OS.isFamilyUnix() == false)
+        {
+            throw new UnsupportedOperationException("video can be converted in unix family OS only");
+        }
+
+        try
+        {
+            File file = new File(fielPath);
+            if (file.exists() == false)
+            {
+                throw new IllegalAccessException("File not found");
+            }
+            CommandLine cmdLine = new CommandLine("ffmpeg");
+            cmdLine.addArgument("-i");
+            cmdLine.addArgument(file.getAbsolutePath());
+            cmdLine.addArgument("-vcodec");
+            cmdLine.addArgument("libx264");
+            cmdLine.addArgument("-crf");
+            cmdLine.addArgument("20");
+            // experimentaloption but what can you do
+            cmdLine.addArgument("-strict");
+            cmdLine.addArgument("-2");
+            //
+            String fileName = file.getName();
+            if (fileName.indexOf(".") > 0)
+            {
+                fileName = fileName.substring(0, fileName.lastIndexOf("."));
+            }
+            File outputFile = new File(outputDirectory + "/" + String.format("%s_web.mp4", fileName));
```

Comment:
Make the output file naming configurable, e.g. by another command line parameter.